web3.js/src/connection.ts

```diff
@@ -4723,14 +4825,25 @@ export class Connection {
     callback: SignatureSubscriptionCallback,
     options?: SignatureSubscriptionOptions,
   ): ClientSubscriptionId {
-    const clientSubscriptionId = this._nextClientSubscriptionId++;
-    this._signatureSubscriptions[clientSubscriptionId] = {
-      signature,
-      callback,
-      options,
-      subscriptionId: null,
-    };
-    this._updateSubscriptions();
+    const clientSubscriptionId = this._makeSubscription({
+      callback: (notification, context) => {
+        callback(notification, context);
+        // Signatures subscriptions are auto-removed by the RPC service
+        // so no need to explicitly send an unsubscribe message.
```

Comment:
this comment implies that we won't send the unsubscribe message, but it looks like we *are* doing that now

Reply:
We don't, actually! Check out the special case driven by `_subscriptionsAutoDisposedByRpc`, and check out the ‘auto-disposing subscriptions’ section of the tests.